### PR TITLE
Fix audit unit test file location

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind cleanup

xref https://github.com/kubernetes/kubernetes/issues/99881#issuecomment-855002014

```release-note
NONE
```

/assign @BenTheElder

### CODE REVIEW

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve
thank you!

File: staging/src/k8s.io/apiserver/pkg/server/options/audit_test.go [hunk 1]
```diff
@@ -33,6 +34,9 @@ import (
 )
 
 func TestAuditValidOptions(t *testing.T) {
+	tmpDir := t.TempDir()
+	auditPath := filepath.Join(tmpDir, "audit")
+
```
Comment (libaoan) [inline staging/src/k8s.io/apiserver/pkg/server/options/audit_test.go:39]: here, you need `defer os.rRemove(auditPath)`

File: staging/src/k8s.io/apiserver/pkg/server/options/audit_test.go [hunk 2]
```diff
@@ -148,30 +152,33 @@ func TestAuditValidOptions(t *testing.T) {
 }
 
 func TestAuditInvalidOptions(t *testing.T) {
+	tmpDir := t.TempDir()
+	auditPath := filepath.Join(tmpDir, "audit")
```
Comment (libaoan) [inline staging/src/k8s.io/apiserver/pkg/server/options/audit_test.go:156]: and here...
`defer os.rRemove(auditPath)`

**Review by libaoan — COMMENTED** [review]
need remove the tmpfile

File: staging/src/k8s.io/apiserver/pkg/server/options/audit_test.go (hunk 1 above)
Comment (liggitt) [reply staging/src/k8s.io/apiserver/pkg/server/options/audit_test.go:39]: t.TempDir is automatically removed when the test ends

**Comment by BenTheElder** [comment]
For the record:
https://golang.org/pkg/testing/#T.TempDir

> TempDir returns a temporary directory for the test to use. The directory is automatically removed by Cleanup when the test and all its subtests complete. Each subsequent call to t.TempDir returns a unique directory; if the directory creation fails, TempDir terminates the test by calling Fatal.



And if you check the implementation you can see it does `os.RemoveAll(tempDir)` so you don't need to cleanup any of the contents manually, it's already deferred to the scope of the `testing.T` by the stdlib.
https://golang.org/src/testing/testing.go?s=31462:31495#L946

**Comment by BenTheElder** [comment]
this PR is blocked by https://github.com/kubernetes/test-infra/issues/22432

**Review by cbandy — APPROVED** [review]

**Comment by liggitt** [comment]
Did a new required presubmit just get added?

/test pull-kubernetes-node-crio-e2e

**Comment by harche** [comment]
@liggitt `pull-kubernetes-node-crio-e2e` started running into infrastructure related issues after making it blocking. Here is PR to make it non-blocking again. https://github.com/kubernetes/test-infra/pull/22436

**Pull request merged**